ctt-frontend/src/app/services/aws-category.service.ts

```diff
@@ -1,56 +1,113 @@
 import { Injectable } from '@angular/core';
 import { Category } from '../interfaces/category';
-import { StatisticType } from '../interfaces/statistics';
+import { APIService } from './API.service';
 import { CategoryService, CreateCategoryInput } from './category.service';
+import { CreateCategoryInput as APICreateInput } from './API.service';
+import { UpdateCategoryInput as APIUpdateInput } from './API.service';
+import { StatisticType as APIStatisticType } from './API.service';
+import { StatisticType } from '../interfaces/statistics';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AwsCategoryService extends CategoryService {
+export class AwsCategoryService implements CategoryService {
+  constructor(private api: APIService) {}
+
   create(input: CreateCategoryInput): Promise<Category> {
-    return Promise.resolve({
-      color: '#123455',
-      name: 'sleep',
-      id: 'a',
-      reminderInterval: 10,
-      excludeFromStatistics: [StatisticType.absoluteTime],
-    });
+    const i: APICreateInput = {
+      name: input.name,
+      color: input.color,
+      reminderInterval: input.reminderInterval,
+      excludeFromStatistics: input.excludeFromStatistics.map((v) =>
+        v === StatisticType.absoluteTime
+          ? APIStatisticType.AbsoluteTime
+          : APIStatisticType.RelativeTime
+      ),
+    } as APICreateInput;
+    return this.api.CreateCategory(i).then(
+      (result) =>
+        Promise.resolve({
+          color: result.color,
+          id: result.id,
+          name: result.name,
```

Comment:
I think also private methods for converting an `Category` to an `ApiCategory` and the other way make sense.